### PR TITLE
move homepage to code

### PR DESCRIPTION
**Pull request by xithan**
also
- deploy task: mark PT stories as deployed
- remove deprecated `.default-btn`
- filter for "add metric to formula" list
- support more Wolfram Language functions

### CODE REVIEW

File: lib/capistrano/tasks/deploy.cap [hunk 1]
```diff
@@ -67,13 +67,45 @@ namespace :deploy do
         env = "env RAILS_ENV=production"
         num_jobs = fetch :num_delayed_job_processes
         execute :rvm, "default do", env, "./bin/delayed_job -n", num_jobs, "restart"
+    end
       end
     end
-  end
```
[the remark this answers is not in the document]
Comment (xithan) [reply lib/capistrano/tasks/deploy.cap]: fixed in the next PR

File: lib/capistrano/tasks/deploy.cap [hunk 2]
```diff
@@ -67,13 +67,45 @@ namespace :deploy do
...
+      puts "Searching for #{story.id} in local git repo."
+      search_result = `git log --grep #{story.id} #{staging_deploy_tag}`
+      if search_result.length > 0
+        puts "Found #{story.id}, marking as delivered."
+        story.notes.create(:text => "Delivered by staging deploy script.")
+        story.update({"current_state" => "delivered"})
+      else
+        puts "Coult not find #{story.id} in git repo."
+      end
+    end
```
[the remark this answers is not in the document]
Comment (xithan) [reply lib/capistrano/tasks/deploy.cap]: I wasn't aware that we do methods in cap files

File: mod/homepage/set/right/community.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  def multi_card_editor?
+    true
+  end
+
+  def nested_fields_for_edit
+    [[:title],
+     [:icon],
+     [:info]]
+  end
+end
```
[the remark this answers is not in the document]
Comment (xithan) [reply mod/homepage/set/right/community.rb]: But we need a better styling for nested editors. The edit page for the homepage looks very messy.

File: mod/homepage/set/self/homepage_communities.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+include_set Abstract::HamlFile
+
+format :html do
+  COMMUNITIES = ["NGOs", "Researchers", "Education", "Companies",
+                      "Standard Bodies", "Volunteers", "Investors", "Press"].freeze
```
[the remark this answers is not in the document]
Comment (xithan) [reply mod/homepage/set/self/homepage_communities.rb]: should be but I was too lazy

File: lib/capistrano/tasks/deploy.cap (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethn) [reply lib/capistrano/tasks/deploy.cap]: I'm getting dogmatic about breaking up code in my old age.  My old brain can't handle more than 5 lines of code at a time.

**Pull request merged**